core/src/main/java/io/questdb/griffin/engine/functions/eq/ContainsEqIPv4Utils.java

```diff
@@ -0,0 +1,174 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2023 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.eq;
+
+import io.questdb.cairo.sql.Function;
+import io.questdb.cairo.sql.Record;
+import io.questdb.cairo.sql.SymbolTableSource;
+import io.questdb.griffin.PlanSink;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.BinaryFunction;
+import io.questdb.griffin.engine.functions.BooleanFunction;
+import io.questdb.griffin.engine.functions.UnaryFunction;
+import io.questdb.std.NumericException;
+
+import static io.questdb.std.Numbers.IPv4_NULL;
+import static io.questdb.std.Numbers.getIPv4Subnet;
+
+public class ContainsEqIPv4Utils {
+
+    private ContainsEqIPv4Utils() {
+    }
+
+    public static Function containsIPv4(Function ipv4Func, Function strFunc, int strFuncPosition) throws SqlException {
+        if (strFunc.isConstant()) {
+            CharSequence constValue = strFunc.getStrA(null);
+            if (constValue == null) {
+                return new NullCheckFunc(ipv4Func);
```

Comment:
Strictly speaking, this is a breaking change, but I agree that this would be more logical and users won't notice the change. Applied the suggestion in 257aeeb4bb0ebb5d21a9586bb6a0b709aaf993c8